java-frontend/src/test/files/se/DivisionByZeroCheck.java

```diff
@@ -0,0 +1,170 @@
+class A {
+  void foo(int r) {
+    int z1 = 0;
+    int z2 = z1;
+    r = 1 / z2; // Noncompliant {{Make sure 'z2' can't be zero before doing this division.}}
+  }
+
+  void zug(int r) {
+    int z1 = 0x0;
+    int z2 = 15;
+    z2 *= z1;
+    r = 1 / z2; // Noncompliant {{Make sure 'z2' can't be zero before doing this division.}}
+  }
+
+  void zup(int r) {
+    int z1 = 0x1;
+    int z2 = 15;
+    z2 *= z1;
+    r = 1 / z2; // Compliant
+  }
+
+  void rug(int r) {
+    int z1 = 0;
+    int z2 = 15;
+    r = 1 / (z2 * z1); // Noncompliant [[sc=13;ec=22]] {{Make sure this expression can't be zero before doing this division.}}
+  }
+
+  int pdf(int p) {
+    int r = 0;
+    r = -r;
+    return p / r; // Noncompliant
+  }
+
+  int jac(int p) {
+    if (p == 0) {
+      return 0;
+    }
+
+    if (p < 0) {
+      p = -p;
+    }
+
+    if (p == 1) {
+      return 1;
+    }
+
+    int u = 14;
+    u %= p; // Compliant
+    return 0;
+  }
+
+  int car(int s) {
+    if (s >= 0) {}
+
+    if (s <= 0) {
+    } else {
+      int x = 14 / s; // Noncompliant - False Positive !!
```

Comment:
I'm still working on this issue, which corresponds to cases remaining in ruling JDK6.